pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -706,16 +720,20 @@ func (r *ReconcileInfinispan) scheduleUpgradeIfNeeded(infinispan *infinispanv1.I
 
 	// If the operator's default image differs from the pod's default image,
 	// schedule an upgrade by gracefully shutting down the current cluster.
-	if podDefaultImage != desiredImage {
-		if err := r.update(infinispan, func() {
-			logger.Info("schedule an Infinispan cluster upgrade", "pod default image", podDefaultImage, "desired image", desiredImage)
-			infinispan.SetCondition(infinispanv1.ConditionUpgrade, metav1.ConditionTrue, "")
-			infinispan.Spec.Replicas = 0
-		}); err != nil {
-			return &reconcile.Result{}, err
-		}
+	return podDefaultImage != desiredImage, nil
+}
+
+func IsUpgradeRequired(infinispan *infinispanv1.Infinispan) (bool, error) {
+	podList, err := PodList(infinispan)
+	if err != nil {
+		return false, err
 	}
-	return nil, nil
+	return upgradeRequired(infinispan, podList)
+}
+
+func PodList(infinispan *infinispanv1.Infinispan) (*corev1.PodList, error) {
+	podList := &corev1.PodList{}
+	return podList, kubernetes.ResourcesList(infinispan.Namespace, PodLabels(infinispan.Name), podList)
```

Comment:
seems to we need execute `err := kubernetes.ResourcesList(infinispan.Namespace, PodLabels(infinispan.Name), podList)` and then `return podList, err`
May be I'm wrong

Reply:
I don't think it matters, as `podList` is just a pointer. The call to `ResourcesList` updates the value associated with the pointer.